lib/matplotlib/quiver.py

```diff
@@ -251,47 +256,154 @@ def __init__(self, Q, X, Y, U, label,
         """
         super().__init__()
         self.Q = Q
-        self.X = X
-        self.Y = Y
         self.U = U
         self.angle = angle
         self.coord = coordinates
         self.color = color
-        self.label = label
         self._labelsep_inches = labelsep
-        self.labelsep = (self._labelsep_inches * Q.axes.figure.dpi)
 
         # try to prevent closure over the real self
         weak_self = weakref.ref(self)
 
         def on_dpi_change(fig):
             self_weakref = weak_self()
             if self_weakref is not None:
-                self_weakref.labelsep = self_weakref._labelsep_inches * fig.dpi
                 # simple brute force update works because _init is called at
                 # the start of draw.
                 self_weakref._initialized = False
 
         self._cid = Q.axes.figure.callbacks.connect(
             'dpi_changed', on_dpi_change)
 
-        self.labelpos = labelpos
-        self.labelcolor = labelcolor
+        _api.check_in_list(['N', 'S', 'E', 'W'], labelpos=labelpos)
+        self._labelpos = labelpos
         self.fontproperties = fontproperties or dict()
-        self.kw = kw
+        self.kw = kwargs
         _fp = self.fontproperties
-        # boxprops = dict(facecolor='red')
-        self.text = mtext.Text(
-            text=label,  # bbox=boxprops,
-            horizontalalignment=self.halign[self.labelpos],
-            verticalalignment=self.valign[self.labelpos],
-            fontproperties=font_manager.FontProperties._from_any(_fp))
-
-        if self.labelcolor is not None:
-            self.text.set_color(self.labelcolor)
+        self._text = mtext.Text(
+            x=X, y=Y, text=label,
+            horizontalalignment=self.halign[self._labelpos],
+            verticalalignment=self.valign[self._labelpos],
+            fontproperties=font_manager.FontProperties._from_any(_fp),
+            color=labelcolor)
+
         self._initialized = False
         self.zorder = Q.zorder + 0.1
 
+    def get_x(self):
```

Comment:
Perhaps we can do without {get,set}_{x,y} and just stick to get_position (as for Text)?  the backcompat X and Y would be kept for backcompat but can use straight lambdas (`X = property(lambda self: self._text.get_position()[0], self.text.set_x)` -- setting `stale` shouldn't really matter here as Text will propagate staleness to the parent Axes anyways)

Reply:
Agree. We only have get_x on Rectangle and FancyBboxPatch. Let's keep the API size down and only provide get/set_position.